### ISSUE TITLE
Fix Vite HMR

### ISSUE THREAD

**Issue by dmmulroy** [dmmulroy/create-melange-app#34]
We use: https://github.com/pdelacroix/vite-plugin-melange

**Comment by dmmulroy** [comment]
Okay, definitely an issue with the dune files/vite config in our project. HMR is working fine using https://github.com/pdelacroix/melange-vite-template

**Comment by dmmulroy** [comment]
Actually now it seems I can't get HMR to work w/ that template 🤔

**Comment by dmmulroy** [comment]
It appears to be a problem with `chokidar` and `osx`. On some projects, updating `vite.config.js` to use `usePolling` seems to resolve issues.

```js
import { defineConfig } from "vite";
import melangePlugin from "vite-plugin-melange";

export default defineConfig({
  plugins: [
    melangePlugin({
      emitDir: "src",
      buildCommand: "opam exec -- dune build @react",
      watchCommand: "opam exec -- dune build --watch @react",
    }),
  ],
  server: {
    watch: {
      usePolling: true,
      /* awaitWriteFinish: {
        stabilityThreshold: 500,
        pollInterval: 20,
      }, */
    },
  },
});
```

**Comment by dmmulroy** [comment]
Configuring `chokidar` has proved to be quite a tricky and finicky process with `vite-melange-app` with multiple users (myself included) having mixed results.

In the interim of completing #37 we're going to utilize `vite` in the following way:

`./dune`

```
(dirs :standard \ node_modules)

(alias
 (name all)
 (deps
  (alias_rec vite)))

(rule
 (alias vite)
 (targets
  (dir dist))
 (deps
  (alias_rec app_name)
  (:vite ./vite.config.js)
  (:index_html ./index.html))
 (action
  (system "../../node_modules/.bin/vite build"))
 (mode
  (promote (until-clean))))

(melange.emit
 (target output)
 (alias app_name)
 (libraries app)
 (module_systems
  (es6 mjs)))

```

`./vite.config.js`

```js
import { nodeResolve } from "@rollup/plugin-node-resolve";

export default {
  build: {
    outDir: "./dist",
  },
  plugins: [nodeResolve()],
};
```

`src/dune`
```
(library
 (name app)
 (modes melange)
 (libraries bindings create_melange_app reason-react)
 (preprocess
  (pps melange.ppx reason-react-ppx)))
```

`./package.json`

```json
{
  "name": "app_name",
  "version": "0.0.1",
  "description": "Enter a description of your app here",
  "scripts": {
    "dev": "dune build && concurrently 'npm:vite-dev' 'npm:dune-watch'",
    "vite-dev": "vite",
    "build": "dune build",
    "serve": "dune build && vite preview"
  },
  "dependencies": {
    "@rollup/plugin-node-resolve": "^15.2.3",
    "react": "^18.0.0",
    "react-dom": "^18.0.0"
  },
  "devDependencies": {
    "concurrently": "^8.2.2",
    "vite": "^4.5.0",
    "vite-plugin-melange": "^2.2.0"
  }
}
```

**Comment by dmmulroy** [comment]
Resolved by: 69d660f
